fc/mef.py

```diff
@@ -736,230 +506,223 @@ def get_transform_fxn(data_beads, peaks_mef, mef_channels,
-                                                        **find_peaks_params)
-                                    for di in data_clustered])
+        if verbose:
+            print("({}) Step 2: Population Statistic".format(mef_channel))
+            print("  Fluorescence per population (Channel Units):")
+            print("    " + str(stats_values))
+
+        ###
+        # 3. Select populations to be used for fitting
+        ###
+
+        # Select populations based on selection_func
+        if selection_func is not None:
+            selected_mask = selection_func([pi for pi in populations_channel],
+                                           **selection_params)
         else:
-            raise ValueError("Peak finding method {} not recognized."
-                .format(find_peaks_method))
-        # Sort peaks and clusters
-        ind_sorted = np.argsort(peaks_ch)
-        peaks_sorted = peaks_ch[ind_sorted]
-        data_sorted = [data_clustered[i] for i in ind_sorted]
+            selected_mask = np.ones(n_clusters, dtype=bool)
+
+        # Discard values specified as nan in mef_values_channel
+        selected_mask = np.logical_and(selected_mask,
+                                       ~np.isnan(mef_values_channel))
+
+        # Get selected channel and mef values
+        selected_channel = stats_values[selected_mask]
+        selected_mef = mef_values_channel[selected_mask]
 
         # Accumulate results
-        if full:
-            peaks_ch_all.append(peaks_ch)
-            if find_peaks_method == 'smoothed_mode':
-                peaks_hists_all.append(hists_smooth)
+        if full_output:
+            selected_channel_all.append(selected_channel)
+            selected_mef_all.append(selected_mef)
+
         # Print information
         if verbose:
-            print("- STEP 2. PEAK IDENTIFICATION.")
-            print("Channel peaks identified:")
-            print(peaks_sorted)
+            print("({}) Step 3: Population Selection".format(mef_channel))
+            print("  {} populations selected.".format(len(selected_channel)))
+            print("  Fluorescence of selected populations (Channel Units):")
+            print("    " + str(selected_channel))
+            print("  Fluorescence of selected populations (MEF Units):")
+            print("    " + str(selected_mef))
+
         # Plot
         if plot:
-            # Get colors for peaks
-            colors = [fc.plot.cmap_default(i)\
-                                for i in np.linspace(0, 1, n_clusters)]
+            # Get colors for each population. Colors are taken from the default
+            # colormap in fc.plot, if the population has been selected.
+            # Otherwise, the population is displayed in gray.
+            color_levels = np.linspace(0, 1, n_clusters)
+            colors = [fc.plot.cmap_default(level)
+                      if selected else (0.6, 0.6, 0.6)
+                      for selected, level in zip(selected_mask, color_levels)]
```

Comment:
this is super small, but i'd put `if selected else (0.6, 0.6, 0.6)` on the same line as `fc.plot.cmap_default(level)` if it fits, otherwise I'd tab it in one, because those two statements are part of the same clause (the value which goes into the list; it's not a `for` clause and it's not a real `if` clause because it's the weird ternary `if`-`else` thing).